src/plugins/intel_cpu/src/infer_request.cpp

```diff
@@ -285,565 +320,456 @@ void InferRequestBase::changeDefaultPtr() {
+const ov::Output<const ov::Node>& SyncInferRequest::get_internal_port(const ov::Output<const ov::Node>& port) const {
+    auto name = get_port_name(port, m_is_legacy_api);
+    bool is_input = ov::op::util::is_parameter(port.get_node());
+    if (is_input) {
+        return m_input_ports_map[name];
+    } else {
+        return m_output_ports_map[name];
     }
+}
 
-    if (!data)
-        IE_THROW(NotAllocated) << "Failed to set empty blob with name: \'" << name << "\'";
-
-    bool isInput = false;
-    const auto inputNodeItr = modelInputsMap.find(name);
-    const auto outputNodeItr = modelOutputsMap.find(name);
-
-    if (inputNodeItr != modelInputsMap.end()) {
-        if (!inputNodeItr->second) {
-            IE_THROW() << "Can't set blob with name: " << name << ", because has null pointer to input node";
-        }
-        isInput = true;
-    } else if (outputNodeItr != modelOutputsMap.end()) {
-        if (!outputNodeItr->second) {
-            IE_THROW() << "Can't set blob with name: " << name << ", because has null pointer to output node";
+void SyncInferRequest::set_tensor(const ov::Output<const ov::Node>& in_port, const ov::Tensor& in_tensor) {
+    OV_ITT_SCOPED_TASK(itt::domains::intel_cpu, "set_tensor");
+    if (!in_tensor)
+        OPENVINO_THROW("Failed to set empty tensor for port!");
+    auto is_compiled_model_port = check_compiled_model_port(in_port);
+    auto port = get_internal_port(in_port);
+    auto tensor = in_tensor;
+
+    // WA: legacy api create blob with ANY layout will not set BlockingDesc, which will lead to tensor.get_shape()
+    // return empty shape but tensor.get_size() return correct value, and tensor.reshape() cannot update
+    // BlockingDesc, so to construct new tensor with original tensor's data, which is only for ov legacy api usage.
+    if (in_port.get_partial_shape().is_static() && in_tensor.get_size() > 0 && in_tensor.get_shape().size() == 0 &&
+        in_tensor.get_size() == ov::shape_size(in_port.get_shape()) && in_port.get_shape().size() > 0) {
+        tensor = ov::Tensor(in_tensor.get_element_type(), in_port.get_shape(), in_tensor.data());
+    }
+    auto name = get_port_name(in_port, m_is_legacy_api);
+    auto is_precision_changed = m_port_precision_changed[name];
+
+    // Precision has been changed
+    if (is_precision_changed) {
+        if (!is_compiled_model_port) {
+            // Original port
+            auto _orig_port = m_orig_ports_map[name];
+            if (_orig_port.get_element_type() == in_tensor.get_element_type()) {
+                // Original port + original port's tensor
+                m_aux_tensors[name] = in_tensor;
+                tensor = ov::ISyncInferRequest::get_tensor(port);
+                tensor.set_shape(in_tensor.get_shape());
+            } else if (port.get_element_type() == in_tensor.get_element_type()) {
+                // Original port + compiled model port's tensor
+                tensor = in_tensor;
```

Comment:
This actually a redundant code since on L490 `tensor = in_tensor` was already performed.

Reply:
yes, you are right